### ISSUE TITLE
Something wrong when self.n_anchors > 1

### ISSUE THREAD

**Issue by YuanXiaode** [Megvii-BaseDetection/YOLOX#671]
作者好，时间紧就中文说了，我注意到几个检测头的输出维度处理不一样，self.reg_preds的输出没有乘上self.n_anchors。
            self.cls_preds.append(
                nn.Conv2d(
                    in_channels=int(256 * width),
                    out_channels=self.n_anchors * self.num_classes,
                    kernel_size=1,
                    stride=1,
                    padding=0,
                )
            )
            self.reg_preds.append(
                nn.Conv2d(
                    in_channels=int(256 * width),
                    out_channels=4,
                    kernel_size=1,
                    stride=1,
                    padding=0,
                )
            )
            self.obj_preds.append(
                nn.Conv2d(
                    in_channels=int(256 * width),
                    out_channels=self.n_anchors * 1,
                    kernel_size=1,
                    stride=1,
                    padding=0,
                )
            )
根据后面对output的处理：
        output = output.view(batch_size, self.n_anchors, n_ch, hsize, wsize)
        output = output.permute(0, 1, 3, 4, 2).reshape(
            batch_size, self.n_anchors * hsize * wsize, -1
        )
这里self.reg_preds的输出必须乘上 self.n_anchors，才能给view成代码的维度。

另外这句：
        grid = grid.view(1, -1, 2)
        output[..., :2] = (output[..., :2] + grid) * stride
        output[..., 2:4] = torch.exp(output[..., 2:4]) * stride
grid的维度是(1,H x W, 2)，而 output[..., :2]的维度是 (batch_size, self.n_anchors x H x W,2 )，这里第二个维度，就算用广播原理，也是无法相加的。

**Comment by Joker316701882** [comment]
yolox是anchor free方法，n_anchors默认为1，是冗余的概念。这是我们做ablation时候用到的参数，使用yolox的时候不需要考虑n_anchors不为1的情况，我们后续会清一下这部分代码

**Comment by YuanXiaode** [comment]
了解了，感谢作者回复，非常荣幸能看到这么好的项目
